roles/activation_keys/tasks/main.yml

```diff
@@ -21,6 +21,6 @@
     purpose_usage: "{{ item.purpose_usage | default(omit) }}"
     purpose_role: "{{ item.purpose_role | default(omit) }}"
     purpose_addons: "{{ item.purpose_addons | default(omit) }}"
-    state: present
-  with_items:
+    state: "{{ item.state | default('present') }}"
+  loop:
```

Comment:
If you keep "with_items" things will work again.
"loop" is stricter about the syntax

Reply:
Replaced loop with with_items